### PR TITLE
add plot_hpd

### PR DESCRIPTION
**Pull request by aloctavodia**
It turns out that #378 was not a good general solution: 
* Only works when point are more or less uniformly distributed.
* The user needs to take extra steps to actually plot the HPD.

 This reverts #378 and introduces a new plotting function `plot_hpd` that by defaults perform smoothing. The user has full control over the smoothing parameters by using the `smooth_kwargs`. It also has a color argument. I am assuming this argument will be the most commonly used thus it is better to have it accessible instead of asking the user to specify it using `plot_kwargs`.

Some examples 

with smoothing (left) without (right)
reg_hpd

without smoothing
b11197_07_15

with smoothing
b11197_07_15b

Probably I should also add tests and docs, but first I want to know your opinions

### CODE REVIEW

**Comment by canyon289** [comment]
This looks nice. I like that the `stats.hdp` method is simplified, and the addition of a new plot makes things clearer. Making the smoothing more obvious is prudent as well

**Comment by canyon289** [comment]
@aloctavodia I rebased your branch on top of master. Lets see if this works

**Comment by canyon289** [comment]
Whew finally got it right. I have some comments now though :)

File: arviz/tests/test_plots.py [hunk 1]
```diff
@@ -79,8 +80,8 @@ def pymc3_sample_ppc(models):
 
 
 @pytest.fixture(scope="module")
-def data():
-    data = eight_schools_params()
+def data(eight_schools_params):
```
[the remark this answers is not in the document]
Comment (aloctavodia) [reply arviz/tests/test_plots.py]: Not sure about the best route here, shouldn't this be part of another PR?

File: arviz/tests/test_plots.py [hunk 2]
```diff
@@ -404,3 +405,18 @@ def test_plot_compare(models, kwargs):
 
     axes = plot_compare(model_compare, **kwargs)
     assert axes
+
+
+@pytest.mark.parametrize("model_fit", ["pymc3_fit", "stan_fit"])
```
[the remark this answers is not in the document]
Comment (aloctavodia) [reply arviz/tests/test_plots.py]: thanks! And thanks for helping me with the tests

File: arviz/tests/test_plots.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (canyon289) [reply arviz/tests/test_plots.py]: Oh, yea I see this was there before you started. Another PR is fine!

**Comment by canyon289** [comment]
LGTM

**Pull request merged**